### ISSUE TITLE
Switch Network popup

### ISSUE THREAD

**Issue by VivekMakwana24** [RootSoft/walletconnect-dart-sdk#67]
Hello,

Is there a way in which I can provide a SwitchNetwork popup to the user?

just like the below-attached screenshot

![Image from iOS](https://user-images.githubusercontent.com/48508009/182310247-97a03957-2b9d-45e2-ac44-c5f936a6e007.png)

**Comment by VivekMakwana24** [comment]
Does any one have any idea about showing the switch network popup?

**Comment by RootSoft** [comment]
Are you trying to show a popup in your own wallet or call the MetaMask switch network popup?

**Comment by VivekMakwana24** [comment]
@RootSoft I want to call the MetaMask switchNetwork popup,

**Comment by VivekMakwana24** [comment]
@RootSoft Could you please update on this?
Thanks :)

**Comment by VivekMakwana24** [comment]
@RootSoft Any update on my query?

**Comment by RootSoft** [comment]
@VivekMakwana24 I don't think it's possible to do that since that's native to MM, and I don't think they provide an API to show the switch network popup. Your best bet would be to ask in the MM repo?

**Comment by VivekMakwana24** [comment]
Ok let me check with them,
I have found the below link, do you think it will be useful?
**https://community.metamask.io/t/how-to-send-switch-network-request-from-dapp-to-metamask-mobile/13501**

**Comment by VivekMakwana24** [comment]
@RootSoft 
Below is the code that I have tried, but unfortunately I am not getting any response from it.

`  
```
Future<dynamic> switchNetwork() async {
    final request = JsonRpcRequest(
      id: payloadId,
      method: 'wallet_switchEthereumChain',
      params: [
        {
          "chainId": "0xa86a",
        }
      ],
    );

    session.handshakeId = request.id;
    session.handshakeTopic = const Uuid().v4();

    debugPrint("req2 ${request.toJson()}");

    final response = await sendRequest(request, topic: session.handshakeTopic);
    debugPrint("response ${response}");

    return response;
  }
```
`
Request : {id: 1660718671093646, jsonrpc: 2.0, method: wallet_switchEthereumChain, params: [{chainId: 0xa86a}]}
Response : null

**Comment by marqueeeeeee** [comment]
try this, it's working on mine, i just need to know how to update the session

<img width="575" alt="Screen Shot 2022-09-09 at 12 24 47 AM" src="https://user-images.githubusercontent.com/8085652/189175038-faaa1886-3e75-4037-be07-6689326e4ab2.png">

**Comment by robmagario** [comment]
Any updates on this issue? I am also trying to switch the network but no luck.

**Comment by VivekMakwana24** [comment]
@robmagario No, I have tried by contacting the walletConnect community but no solution yet

**Comment by RootSoft** [comment]
@VivekMakwana24 I think you'll have to dig in the Metamask source code. Maybe build the app and step through it.
Unfortunately, at the moment I don't have a lot of time to dig deep into this. This is more related to Metamask then WC in my opinion.

**Comment by olegys** [comment]
@VivekMakwana24 Try this

`      await connector.sendCustomRequest(
        method: 'wallet_switchEthereumChain',
        params: [
          {
            'chainId': '0x${provider.chainId.toRadixString(16)}',
          },
        ],
      );`

**Comment by VivekMakwana24** [comment]
@olegys Did it worked with metamask?

**Comment by olegys** [comment]
> 

@VivekMakwana24  Yes

**Comment by VivekMakwana24** [comment]
@olegys 

This is how I have tried it, But Unfortunately, I am receiving null response.

I tried to call this method before launching the walletURI for connection.

Is it the correct flow?

![Screenshot from 2022-10-04 18-45-11](https://user-images.githubusercontent.com/48508009/193828945-f89feaf2-eb9f-4ba7-a455-e83a3400f312.png)
